core/events/domain/event.py

```diff
@@ -443,34 +463,58 @@ def project(self, submission: Submission) -> Submission:
 class FinalizeSubmission(Event):
     """Send the submission to the queue for announcement."""
 
+    REQUIRED = [
+        'creator', 'primary_classification', 'submitter_contact_verified',
+        'submitter_accepts_policy', 'license', 'source_content', 'metadata',
+    ]
+    REQUIRED_METADATA = ['title', 'abstract', 'authors']
+
     def validate(self, submission: Submission) -> None:
         """Ensure that all required data/steps are complete."""
         if submission.finalized:
             raise InvalidEvent(self, "Submission already finalized")
         if not submission.active:
             raise InvalidEvent(self, "Submision must be active")
-
-        try:
-            assert submission.creator is not None
-            assert submission.primary_classification is not None
-            assert submission.metadata.title is not None
-            assert submission.metadata.abstract is not None
-            assert len(submission.metadata.authors) > 0
-            assert submission.submitter_contact_verified
-            assert submission.submitter_accepts_policy
-            assert submission.license is not None
-            assert submission.source_content is not None
-        except AssertionError as e:
-            raise InvalidEvent(self, "Submission missing required data") from e
+        self._required_fields_are_complete(submission)
 
     def project(self, submission: Submission) -> Submission:
         """Set :prop:`Submission.finalized`."""
         submission.finalized = True
         return submission
 
+    def _required_fields_are_complete(self, submission: Submission) -> None:
```

Comment:
This minor refactor results in better error reporting when something is missing.